windows/src/engine/keyman32/Keyman32.cpp

```diff
@@ -938,13 +955,14 @@ void RefreshKeyboards(BOOL Initialising)
 
   _td->FInRefreshKeyboards = FALSE;
 }
-
 void ReleaseKeyboards(BOOL Lock)
 {
+  OutputThreadDebugString("ReleaseKeyboards");
   PKEYMAN64THREADDATA _td = ThreadGlobals();
-	if(!_td || _td->lpKeyboards) return;
+	if(!_td || !_td->lpKeyboards) return;
```

Comment:
Note the flipped test here. This was the primary cause of leaks. Fixing this showed up another issue in `KeyboardOptions.cpp` where assertions in `FreeKeyboardOptions` would be triggered for keyboards that had never been loaded were being torn down.